app/src/features/apiClient/screens/apiClient/utils.ts

```diff
@@ -526,3 +527,49 @@ export const apiRequestToHarRequestAdapter = (apiRequest: RQAPI.Request): HarReq
 
   return harRequest;
 };
+
+export const filterOutChildrenRecords = (
+  selectedRecords: Set<RQAPI.Record["id"]>,
+  childParentMap: Record<RQAPI.Record["id"], RQAPI.Record["id"]>,
+  recordsMap: Record<RQAPI.Record["id"], RQAPI.Record>
+) =>
+  [...selectedRecords]
+    .filter((id) => !childParentMap[id] || !selectedRecords.has(childParentMap[id]))
+    .map((id) => recordsMap[id]);
+
+export const processRecordsForDuplication = (recordsToProcess: RQAPI.Record[]) => {
+  const recordsToDuplicate: RQAPI.Record[] = [];
+  const queue: RQAPI.Record[] = [...recordsToProcess];
+
+  while (queue.length > 0) {
+    const record = queue.shift()!;
+
+    if (record.type === RQAPI.RecordType.COLLECTION) {
+      const newId = generateDocumentId("apis");
+
+      const collectionToDuplicate: RQAPI.CollectionRecord = Object.assign({}, record, {
+        id: newId,
+        name: `(Copy) ${record.name}`,
+        data: omit(record.data, "children"),
+      });
+
+      recordsToDuplicate.push(collectionToDuplicate);
+
+      if (record.data.children?.length) {
+        const childrenToDuplicate = record.data.children.map((child) =>
+          Object.assign({}, child, { collectionId: newId })
+        );
+        queue.push(...childrenToDuplicate);
+      }
+    } else {
+      const requestToDuplicate: RQAPI.Record = Object.assign({}, record, {
+        id: generateDocumentId("apis"),
+        name: `(Copy) ${record.name}`,
+      });
```

Comment:
the record metadata remains unchanged (and hard to even notice) if we create duplicates using `Object.assign`. 

Shouldn't we use some version of the [original record generator](https://github.com/requestly/requestly/blob/1346e4722e695f3580b33c1b9e517dc9fb19586e/app/src/features/apiClient/screens/apiClient/utils.ts) function here?